charts/valkey/templates/statefulset.yaml

```diff
@@ -0,0 +1,166 @@
+          args: ["sh /scripts/init-cluster.sh & valkey-server /etc/valkey/valkey.conf"]
+          env:
+          - name: POD_IP
+            valueFrom:
+              fieldRef:
+                fieldPath: status.podIP
+          - name: HOSTNAME
+            valueFrom:
+              fieldRef:
+                fieldPath: "metadata.name"
+          - name: REPLICAS
+            value: {{ include "valkey.replicas" . | quote }}
+          - name: NAMESPACE
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.namespace    
+          ports:
+            - name: client
+              containerPort: 6379
+            - name: cluster
+              containerPort: 16379
+          readinessProbe:
+            exec:
+              command:
+              - redis-cli
+              - ping
+            initialDelaySeconds: 10
+            periodSeconds: 5
+          {{- with .Values.resources }}
+          resources:
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
+          volumeMounts:
+            - name: data
+              mountPath: /data
+            - name: workdir
+              mountPath: /etc/valkey
+            - name: scripts
+              mountPath: /scripts
+            {{- with .Values.volumeMounts }}
+              {{- toYaml . | nindent 12 }}
+            {{- end }}
+        {{- if .Values.prometheusExporter.enabled }}
+        - name: redis-exporter
+          image: oliver006/redis_exporter:latest
+          securityContext:
+            runAsUser: 59000
+            runAsGroup: 59000
+            allowPrivilegeEscalation: false
+            capabilities:
+              drop:
+                - ALL
+          resources:
+            requests:
+              cpu: 100m
+              memory: 100Mi
+          ports:
+            - containerPort: {{ .Values.prometheusExporter.port }}
+              name: metrics
+          env:
+            - name: REDIS_ADDR
+              value: redis://localhost:6379
+            - name: REDIS_EXPORTER_IS_CLUSTER
+              value: "true"
+        {{- end }}
+      volumes:
+        - name: config
+          configMap:
+            name: valkey
+        - name: scripts
+          configMap:
+            name: valkey
+            items:
+              - key: init-cluster.sh
+                path: init-cluster.sh
+              - key: init-config.sh
+                path: init-config.sh
+            defaultMode: 0755
+        - name: workdir
+          emptyDir: {}
+      {{- with .Values.volumes }}
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.nodeSelector }}
+      nodeSelector:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.affinity }}
+      affinity:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.tolerations }}
+      tolerations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+  volumeClaimTemplates:
+    - metadata:
+        name: data
```

Comment:
If we keep the original name, we can take over the data for existing deployments. The original was `valkey-data`.

Reply:
Perfect.